### PR TITLE
Automatic merge of refined predictions into human annotations

### LINKED ISSUE #6: Question regarding refining workflow
I'm pretty sure I used this before for refining labels, but now I can't get it to work properly:

If I drag a folder containing a machine labels file, it shows the originally labeled, not extracted points, but all images (loads collected data by default, which is expected). 
If I delete that key points layer and drag in the machine labels, it shows all the frames and only the extracted labels. 
If I refine those and save, it doesn't include the original labels in the output (good thing I backed up that collected data file). So how can I use it to refine labels properly?

Also, in the instructions, I would add a reminder before selecting save layer as as (or ctrl-S) to make sure the key points layer is selected. If the user clicked on the images layer first, does save as, then closes the window, any labeling work during that session will be lost.

### PR DESCRIPTION
**Pull request by jeylau**
@MMathisLab, I think that's more 'elegant' than what I had in mind earlier today:

- if multiple h5 files are found in the folder, they are now all loaded into separate layers; I thought about restricting it to CollectedData_ or machinelabels_ files, but that'd pose problems when loading other files;
- layers are named after the corresponding h5 files for clarity (rather than simply "keypoints");
- upon saving, the filename input in the save dialog does not matter anymore, as the data file will be correctly named automatically;
- when saving a 'machinelabels*' layer, it's either written as a new CollectedData_ file if no other is found, or merge into the existing one like DLC currently does.

I've tested it quick, but I'd be happy to get your feedback 😊

Fixes #6

### CODE REVIEW

**Comment by backyardbiomech** [comment]
A few thoughts:
It basically works for me (all layers loaded, saving works as intended with limits below).

This is on data that has already been merged, so the machine labels rows are already in CollectedData, so not a perfect test (I tested single animal project that is what I have easily available right now). But moved points in machinelabels do correctly overwrite in CollectedData.

After loading, I can hide the h5 layers I'm not interested in (CollectedData and earlier machinelabels). But there is an issue of displaying the name of the selected label in the napari dropdown. When the machinelabels layer is selected, the napari bodypart dropdown doesn't change to the selected body part in the machinelabels layer, it stays stuck on the last selected bodypart in the CollectedData layer. I'm assuming this is just a napari display issue since moving points, displaying text labels on the image in the layer, and saving the data all seem to work.

Also, to throw in what will be a very rare wrench, I have a project where I converted from single animal to multi, so I have `CollectedData_DLCsingleanimal.h5` files which are loaded. And aren't there other files left when running `windows2linux` that might create confusion when loading all h5 files?

In general, with this strategy of loading all files, I can see several problems for users:
1. If someone has extracted 400 frames for a single video (I know, not good, but it happens), but only has to refine 3 frames, they have to jump through all 400 frames to find those 3 to refine. Napari is fast at this, but it could still raise complaints. Is there a way in napari when changing frame to jump to the frame with keypoints in the selected machinelabels layer rather than just the next image?
2. After multiple iterations and extractions, it will get a bit messy having all of the machine labels files in the napari GUI. More layers to manage (hide and/or select to save) means more potential user errors.
3. Thus, instructions will have to be very clear to tell users to hide other layers (including CollectedData) and ensure that only the last machinelabels layer is selected when refining and saving.

**Comment by backyardbiomech** [comment]
It's not quite working. I extracted outliers from videos that had never been labeled (so no existing `CollectedData` files) and refined the labels in napari.

WIth the file save dialogue, I had to enter the name properly. Otherwise it saved it as whatever I entered - no autonaming.

More importantly, the "scorer" line of the saved file was the machine scorer (e.g. `DLC_dlcrnetms5_stinkbugsJan27shuffle2_36000`), not the individual scorer (I always use `DLC`). Also, the liklilhood columns were saved to the output. Unfortunately, I didn't look at the training dataset file, which now has extra columns due to the new "scorer" entries.  I was still "allowed" to create a training set, and train. I didn't catch an error until evaluating the models.

**Comment by MMathisLab** [comment]
@jeylau maybe let's go back to the drawing board on refinement; I suggest a group dev call next week! cc @backyardbiomech will email :)

**Comment by jeylau** [comment]
Hey @backyardbiomech, thanks for the feedback! Are you certain you had pulled the latest changes? I cannot reproduce the behavior you described. In `io.write_hdf`, `filename` (the name you enter in the save dialog) is no longer used, the scorer is either read from CollectedData if present or from the config.yaml, and 'likelihood' is dropped from the machinelabels file 🤔

**Comment by backyardbiomech** [comment]
Ah, yes, it seems to work. Ipython was importing from an `.egg` I had in site-packages rather than from the checked out branch.

**Review by backyardbiomech — APPROVED** [review]
I tested again on an active project with freshly extracted outliers for new videos. That all worked great! I did not test on outliers for a video that already had labeled points.

**Comment by jeylau** [comment]
Awesome! I've just hacked an auto-save feature too, so that the save dialog is no longer shown.

**Comment by backyardbiomech** [comment]
auto-save works great! Thanks, @jeylau !!

**Review by MMathisLab — APPROVED** [review]
